File: 2244-minimum-rounds-to-complete-all-tasks/2244-minimum-rounds-to-complete-all-tasks.py
class Solution:
    def minimumRounds(self, tasks: List[int]) -> int:
        sorted(tasks)
        mp = {};
        ans = 0
        for val in tasks:
            if val not in mp:
                mp[val] = 0;
            mp[val] += 1;
        for key, val in mp.items():
            if (val == 1): return -1;
            ans += math.ceil(val/3)
        return ans;
        
        

File: 2244-minimum-rounds-to-complete-all-tasks/2244-minimum-rounds-to-complete-all-tasks.java
class Solution {
    public int minimumRounds(int[] tasks) {
        HashMap<Integer, Integer> hmMap = new HashMap<>();
        for (int num : tasks) {
            hmMap.put(num, hmMap.getOrDefault(num, 0) + 1);
        }
        Iterator<Integer> it = hmMap.keySet().iterator();
        int ans = 0;
        while (it.hasNext()) {
            int val = hmMap.get(it.next());
            if (val == 1) return -1;
            ans += val / 3;
            if (val % 3 != 0) ans++;
        }
        return (int) ans;
    }
}